Makefile

```diff
@@ -1,50 +1,82 @@
+	docker build -t $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) $(COMMON_DOCKER_PARAMS) .
 
 build-nofeatures: ## Build the container without http requests.
-	docker build -t $(DOCKER_REPO)/$(APP_NAME)_nofeatures:$(VERSION) $(COMMON_DOCKER_PARAMS) --build-arg CARGO_PARAMS="--no-default-features" .
+	docker build -t $(DOCKER_REMOTE_REPO)/$(APP_NAME)_nofeatures:$(VERSION) $(COMMON_DOCKER_PARAMS) --build-arg CARGO_PARAMS="--no-default-features" .
 
 run: ## Run container on port configured in `config.env`
 	docker run -it --rm --env HORUST_LOG=debug -v $(REPO_HOME)/examples/services/longrunning/:/etc/horust/services/ --name="$(NAME)" $(NAME):$(VERSION)
 
 run-bash: ## Run bash with horust
-	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
+	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
 
 stop: ## Stop and remove a running container
 	docker stop $(NAME)
 
-# Docker publish
+# Docker publishing tasks
 publish: build publish-latest publish-version ## publish the `{version}` ans `latest` tagged containers to ECR
 
 publish-latest: tag-latest ## publish the `latest` tagged container
-	@echo 'publish latest to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):latest
+	@echo 'publish latest to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 publish-version: ## publish the `{version}` tagged container to ECR
-	@echo 'publish $(VERSION) to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):$(VERSION)
+	@echo 'publish $(VERSION) to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION)
 
-tag-latest:
+tag-latest: ## tags the latest container with the version listed above
 	@echo 'create tag latest'
-	docker tag $(DOCKER_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REPO)/$(APP_NAME):latest
+	docker tag $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 version: ## output to version
-	@echo $(VERSION)
\ No newline at end of file
+	@echo $(VERSION)
+
+# Docker local development tasks
+## Dargo == Docker Cargo
+dargo-prep: ## This runs everything neccessary to start developing locally in a container
+	@# Spin up a long-running rust container
+	make dargo-run-container
+	# Compile and cache all dependencies
+	@make dargo COMMAND=build
+	@# Perform a first-sweep check, to fill up the cache (it seems to be different than the build one)
+	@make dargo COMMAND=check
+	@# Compile and cache test dependencies, run tests
+	@make dargo COMMAND=test
+	# Now all caches are filled, and subsequent operations will be faster
+	# Consider adding dargo(){ make dargo COMMAND=$1} to your ~/.zshrc or ~/.bashrc for ergonomics
+
+dargo: ## Run a cargo command inside the container
+	@# If the dargo container does not exist, create it
+	@if [ -z $(shell docker ps --format "{{.ID}}" --filter "name=$(LOCAL_DEV_CONTAINER_NAME)") ]; then make dargo-prep; fi
```

Comment:
I'm probably missing something, but in my comment above ^ I was also suggesting to remove the dargo-prep altogether, and call dargo-run-container instead. The cache thing made sense in the previous "version" maybe, in this case if the user either runs cargo test / build /check  which is going to download dependencies anyway. In the same way, when you download any rust project for the first time you don't need to run check / build / test, I think we should just remove the dargo-prep command if there is no good reason for keeping it.

Reply:
Ahh.... yeah I see it now. Replacing